src/java/org/apache/cassandra/index/Index.java

```diff
@@ -341,13 +341,21 @@ default public Callable<?> getPreJoinTask(boolean hadBootstrap)
      * Return true if this index can be built or rebuilt when the index manager determines it is necessary. Returning
      * false enables the index implementation (or some other component) to control if and when SSTable data is
      * incorporated into the index.
-     *
+     * <p>
      * This is called by SecondaryIndexManager in buildIndexBlocking, buildAllIndexesBlocking and rebuildIndexesBlocking
      * where a return value of false causes the index to be exluded from the set of those which will process the
      * SSTable data.
      * @return if the index should be included in the set which processes SSTable data, false otherwise.
      */
-    public boolean shouldBuildBlocking();
+    boolean shouldBuildBlocking();
+    
+    /**
+     * @return true if the index builds SSTable-attached on-disk components
+     */
+    default boolean isSSTableAttached()
```

Comment:
This method allows the SIM to figure out which indexes it should try to build incrementally, and which it should ignore for partial rebuilds actuated via `SSTableAdded` notifications.

TODO: A comment that explains what SSTable-attached indexes are expected to do. In other words, they must be using an `SSTableFlushObserver` to make sure they build during streaming and be able to build SSTable-specific indexes incrementally in the build tasks they provide.

Reply:
A comment would make sense. If SASI or table-based indexes had more future that they have we could even subclass `Index` and have a generic `SSTableAttachedIndex` API, implemented by SAI and SASI. But it probably doesn't worth the additional complexity at this point.

Reply:
Yeah, I think we're all hesitant to make many more changes around this or the index API when we know those two are basically defunct.